src/util/commandUtils.ts

```diff
@@ -0,0 +1,255 @@
+    }
+}
+
+/**
+ * Return the gradle command based on the OS and Terminal for start, start in container, start..
+ */
+export async function getCommandForGradle(buildGradlePath: string, command: string, terminalType?: String, customCommand?: string): Promise<string> {
+    let gradleCmdStart = await gradleCmd(buildGradlePath);
+
+    if (gradleCmdStart === "gradle") {
+        return formDefaultCommand(gradleCmdStart, buildGradlePath, command, "-b=", customCommand);
+    }
+    //checking the OS type for command customization
+    if (isWin()) {
+        return getGradleCommandForWin(gradleCmdStart, buildGradlePath, command, terminalType, customCommand);
+    } else {
+        gradleCmdStart = Path.join(gradleCmdStart, "gradlew");
+        return formDefaultCommand(gradleCmdStart, buildGradlePath, command, "-b=", customCommand);
+    }
+}
+
+/**
+ * Reused from vscode-maven
+ * https://github.com/microsoft/vscode-maven/blob/main/src/mavenTerminal.ts
+ */
+function toDefaultWslPath(p: string): string {
+    const arr: string[] = p.split(":\\");
+    if (arr.length === 2) {
+        const drive: string = arr[0].toLowerCase();
+        const dir: string = arr[1].replace(/\\/g, "/");
+        return `/mnt/${drive}/${dir}`;
+    } else {
+        return p.replace(/\\/g, "/");
+    }
+}
+
+/**
+ * Reused from vscode-maven
+ * https://github.com/microsoft/vscode-maven/blob/2ab8f392f418c8e0fe2903387f2b0013a1c50e78/src/utils/mavenUtils.ts
+ */
+export function isWin(): boolean {
+    return process.platform.startsWith("win");
+}
+
+/**
+ * Returns maven wrapper path or mvn
+ */
+async function mvnCmd(pomPath: string): Promise<string> {
+    const preferMavenWrapper: boolean | undefined = vscode.workspace.getConfiguration("maven").get<boolean>("executable.preferMavenWrapper");
+    if (preferMavenWrapper) {
+        const localMvnwPath: string | undefined = await getLocalMavenWrapper(Path.dirname(pomPath));
+        if (localMvnwPath) {
+            return `${localMvnwPath}`;
+        }
+    }
+    return "mvn";
+}
+
+/**
+ * Returns gradle wrapper path or gradle
+ */
+async function gradleCmd(buildGradle: string): Promise<string> {
+    const preferGradleWrapper: boolean | undefined = vscode.workspace.getConfiguration("java").get<boolean>("import.gradle.wrapper.enabled");
+    if (preferGradleWrapper) {
+        const localGradlewPath: string | undefined = await getLocalGradleWrapper(Path.dirname(buildGradle));
+        if (localGradlewPath) {
+            return `${localGradlewPath}`;
+        }
+    }
+    return "gradle";
+}
+
+/**
+ * Search for potential Maven wrapper, return undefined if does not exist
+ * Reused from vscode-maven
+ * https://github.com/microsoft/vscode-maven/blob/2ab8f392f418c8e0fe2903387f2b0013a1c50e78/src/utils/mavenUtils.ts
+ * @param projectFolder
+ */
+async function getLocalGradleWrapper(projectFolder: string): Promise<string | undefined> {
```

Comment:
Since this method is no longer returning the wrapper, the name should be updated
```suggestion
async function getLocalGradleWrapperDir(projectFolder: string): Promise<string | undefined> {
```